include/elf/arc-reloc.def

```diff
@@ -497,3 +497,24 @@ ARC_RELOC_HOWTO(ARC_NPS_CMEM16, 78, \
                 replace_bits16, \
                 dont, \
                 ( ME ( S + A )))
+
+ARC_RELOC_HOWTO(ARC_JLI_SECTOFF, 63, \
+                2, \
+                10, \
+                replace_jli, \
+                bitfield, \
+                ( ( S - JLI ) ))
+
+ARC_RELOC_HOWTO(ARC_JLI_SECTOFF_SIMM12, 79, \
+                2, \
+                12, \
+                replace_simm12_20, \
+                bitfield, \
+                ( ( ( S - JLI ) >> 2 ) ))
+
+ARC_RELOC_HOWTO(ARC_JLI_32, 80, \
```

Comment:
Thanks, JLI_SECTOFF_SIMM12 is clear to me.
However, it seems to me, this ARC_JLI_32 is also a sort of dynamic reloc. Also, I see you assume JLI_BASE doesn't change its value at run time, as you need it to be constant to solve the ARC_JLI_32 at load time. But then, who stops you to define the value of JLI_BASE when linking and obtaining directly the value which you will anyhow obtain during solving phase of the loader.
Another issue is we need to nicely generate this reloc, rather than spelling it out.